mailbox/tools/indexer/src/test/java/org/apache/mailbox/tools/indexer/ErrorRecoveryIndexationTaskSerializationTest.java

```diff
@@ -0,0 +1,93 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.mailbox.tools.indexer;
+
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.indexer.ReIndexingExecutionFailures;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.server.task.json.JsonTaskSerializer;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
+import org.apache.james.task.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableList;
+
+class ErrorRecoveryIndexationTaskSerializationTest {
+
+    private ReIndexerPerformer reIndexerPerformer;
+    private JsonTaskSerializer taskSerializer;
+    private final String serializedErrorRecoveryReindexingTask = "{\"type\": \"PREVIOUS_FAILURES_INDEXING\"," +
```

Comment:
Shouldn't `PREVIOUS_FAILURES_INDEXING`  be `ErrorRecoveryIndexation` ?

Reply:
(cf Duc's error message above)

Reply:
I think it's good to have the value hardcoded in the test. That way if we change the value of the type, the tests will fail.

Reply:
Nope I mean that's an error... You had harcoded `PREVIOUS_FAILURES_INDEXING`, while the value of the constant is `ErrorRecoveryIndexation`. I'm fine with your previous answer about hardcoding, but then you should replace `PREVIOUS_FAILURES_INDEXING` by `ErrorRecoveryIndexation` :)